packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -366,13 +366,21 @@ export class Summarizer extends EventEmitter implements ISummarizer {
             try {
                 const ack = await this.summaryCollection.waitSummaryAck(refSequenceNumber);
                 refSequenceNumber = ack.summaryOp.referenceSequenceNumber;
-
-                await this.internalsProvider.refreshLatestSummaryAck(
-                    ack.summaryOp.contents.handle,
-                    ack.summaryAck.contents.handle,
-                    refSequenceNumber,
-                    summaryLogger,
-                );
+                this.runningSummarizer.lockedRefreshSummaryAckAction(async () => {
```

Comment:
A comment and some reference to issues it's fixes would be helpful. Otherwise, next engineer will not have a clue why it's here. Also, any chance we can cover it in UTs I.e. removal of this lock would result in test failed, and from test someone can figure out how we can run into this condition)?

Reply:
Yes, today the summarize on demand will hit this on several of the tests.